client/proxy/proxy_wrapper.go

```diff
@@ -254,8 +254,8 @@ func (pw *Wrapper) statusFailedCallback() {
 func (pw *Wrapper) InWorkConn(workConn net.Conn, m *msg.StartWorkConn) {
 	xl := pw.xl
 	pw.mu.RLock()
+	defer pw.mu.RUnlock()
```

Comment:
`phase := pw.Phase`

You can do it like this, without increasing the scope of the lock.